### PR TITLE
Include "location" & "$ref" attribute in SCIM2 responses

### PR DESCRIPTION
**Pull request by sarubi**
## Purpose
Resolves wso2/product-is#5301 and wso2/product-is#5328

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: modules/charon-core/src/main/java/org/wso2/charon3/core/objects/Group.java [hunk 1]
```diff
@@ -156,6 +156,7 @@ public void setMember(String value, String display) throws BadRequestException,
      * @throws BadRequestException
      * @throws CharonException
      */
+    @Deprecated
```
Comment (ruwanta) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/objects/Group.java]: Need to comment why this is deprecated and what are the suggested alternatives

Comment (sarubi) [reply modules/charon-core/src/main/java/org/wso2/charon3/core/objects/Group.java]: Added

File: modules/charon-core/src/main/java/org/wso2/charon3/core/objects/Group.java [hunk 2]
```diff
@@ -208,6 +208,69 @@ private ComplexAttribute setMemberCommon(String userId, String userName, String
...
+            members.setAttributeValue(complexAttribute);
+            this.setAttribute(members);
+        }
+    }
+
+    /**
+     * Create member attribute with three default attributes such as name, value and $ref.
+     *
+     * @param user
+     * @return
```
Comment (ashensw) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/objects/Group.java]: Add a description to @return tag.

Comment (sarubi) [reply modules/charon-core/src/main/java/org/wso2/charon3/core/objects/Group.java]: Addressed with https://github.com/wso2/charon/pull/253

**Pull request merged**